core/sequencer/server.go

```diff
@@ -68,13 +68,27 @@ func createMetrics(mf monitoring.MetricFactory) {
 		domainIDLabel)
 }
 
+// SourcesEntry is a map of SourceSlices by logID.
+type SourcesEntry map[int64]*spb.MapMetadata_SourceSlice
+
+// Watermarks is a map of watermarks by logID.
+type Watermarks map[int64]int64
+
 // LogsReader reads messages in multiple logs.
 type LogsReader interface {
-	// HighWatermarks returns the highest primary key for each log in the mutations table.
-	HighWatermarks(ctx context.Context, domainID string) (map[int64]int64, error)
-	// ReadLog returns the messages in the (low, high] range stored in the specified log.
-	// ReadLog does NOT delete messages.
-	ReadLog(ctx context.Context, domainID string, logID, low, high int64) ([]*mutator.LogMessage, error)
+	// highwatermarks returns the number of items and the highest primary
+	// key up to batchsize items after start (exclusive)
+	HighWatermark(ctx context.Context, domainID string, logID, start int64, batchSize int32) (int32, int64, error)
```

Comment:
I feel like giving names to the output variables here (at least in the interface) would help to avoid any confusion later.